### PR TITLE
Allow single hyphen arguments in bash build

### PR DESCRIPTION
**Pull request by ViktorHofer**
React to the following change incoming from arcade: https://github.com/dotnet/arcade/pull/2127. Will update this PR with the darc update later.

cc @jkotas

### CODE REVIEW

**Comment by ViktorHofer** [comment]
@sandreenko related to your ask (https://github.com/dotnet/corefx/pull/34385#issuecomment-466203479), it would be great if you could change the command to use -os and -arch after this went into master.

**Comment by ViktorHofer** [comment]
cc @safern

**Comment by safern** [comment]
Should we also update corefx-base to benefit from this? i.e: not have conditions for common arguments, and we now can use -framework rather than short -f which is not as descriptive.

https://github.com/dotnet/corefx/blob/master/eng/pipelines/corefx-base.yml#L110

**Comment by ViktorHofer** [comment]
Yeah I plan to do that as part of this PR :)

File: eng/pipelines/corefx-base.yml [hunk 1]
```diff
@@ -84,36 +84,27 @@ jobs:
...
           - _buildScript: ${{ job.buildScriptPrefix }}./build.sh
-          - _args: --restore --build --configuration $(_BuildConfig) --ci --buildtests --arch $(_architecture)
-          - _commonArguments: $(_args)
-          - ${{ if eq(parameters.isOfficialBuild, 'true') }}:
-            - _commonArguments: $(_args) --stripSymbols
-
           - _msbuildCommand: ${{ job.buildScriptPrefix }}./eng/common/msbuild.sh --warnaserror false --ci
-          - _windowsOfficialBuildArguments: ''
+          - ${{ if eq(parameters.isOfficialBuild, 'true') }}:
+            - _commonArguments: $(_commonArguments) -stripSymbols
```
Comment (safern) [inline eng/pipelines/corefx-base.yml]: Is this supported? I didn’t know appending to the same variable in yaml (azure devops) was supported.

Comment (ViktorHofer) [reply eng/pipelines/corefx-base.yml]: Seems like you are reviewing an old commit.

**Comment by ViktorHofer** [comment]
Queued an official build: https://dnceng.visualstudio.com/internal/_build/results?buildId=109941. @safern any idea why the uapaot legs fail in the internal tools restore step with 401?

**Comment by ViktorHofer** [comment]
> @safern any idea why the uapaot legs fail in the internal tools restore step with 401?

Seems like it's not consistently but frequently failing with 401. We should really get the TestILC package to dnceng and get away with the secret.

cc @ericstj @zamont

File: eng/pipelines/corefx-base.yml (hunk 1 above)
Comment (safern) [reply eng/pipelines/corefx-base.yml]: That's what happens when you review from the phone 😄

File: eng/pipelines/corefx-base.yml [hunk 2]
```diff
@@ -84,36 +84,28 @@ jobs:
...
           - group: DotNet-Blob-Feed
 
-        - ${{ if eq(parameters.isOfficialBuild, 'false') }}:
-          - _msbuildCommonParameters: ''
-
         - ${{ if eq(job.submitToHelix, 'true') }}:
           - group: DotNet-HelixApi-Access
           - _archiveTestsParameter: /p:ArchiveTests=Tests
 
+        - _args: -restore -build -configuration $(_BuildConfig) -ci -buildtests -arch $(_architecture) -framework $(_framework) $(_archiveTestsParameter)
```
Comment (safern) [inline eng/pipelines/corefx-base.yml]: Can we do this multiline? for readability purpose? Just like the call to the build script later on this file

Comment (ViktorHofer) [reply eng/pipelines/corefx-base.yml]: sure, if that's the only nit I would prefer do that in a follow-up PR (i.e. codecoverage)

File: eng/pipelines/corefx-base.yml [hunk 3]
```diff
@@ -84,36 +84,28 @@ jobs:
...
+        - _commonArguments: $(_args)
+        - _windowsOfficialBuildArguments: ''
```
Comment (safern) [inline eng/pipelines/corefx-base.yml]: Can you move this to where you're setting `_msbuildCommonParameters: ''` as well?

**Comment by ViktorHofer** [comment]
Thanks @safern. Let's address them in the follow-up PR.

**Pull request merged**